src/utils.ts

```diff
@@ -5,26 +5,29 @@ import * as gitHubHelper from './gitHubHelper';
 import * as inputHelper from './inputHelper';
 import * as trivyHelper from './trivyHelper';
 import * as fileHelper from './fileHelper';
+import { GitHubClient } from './githubClient';
+import { StatusCodes } from "./httpClient";
 
-export function getCheckRunPayloadWithScanResult(trivyStatus: number, dockleStatus: number): any {
-  const headSha = gitHubHelper.getHeadSha();
-  const checkConclusion = getCheckConclusion(trivyStatus, dockleStatus);
-  const checkSummary = getCheckSummary(trivyStatus, dockleStatus);
-  const checkText = getCheckText(trivyStatus, dockleStatus);
-
-  const checkRunPayload = {
-    head_sha: headSha,
-    name: `[container-scan] ${inputHelper.imageName}`,
-    status: "completed",
-    conclusion: checkConclusion,
-    output: {
-      title: "Container scan result",
-      summary: checkSummary,
-      text: checkText
-    }
+export async function createScanResult(trivyStatus: number, dockleStatus: number): Promise<void> {
+  const gitHubClient = new GitHubClient(process.env.GITHUB_REPOSITORY, inputHelper.githubToken);
+  const scanResultPayload = getScanResultPayload(trivyStatus, dockleStatus);
+  const response = await gitHubClient.createScanResult(scanResultPayload);
+  
+  if (response.statusCode == StatusCodes.UNPROCESSABLE_ENTITY
+    && response.body
+    && response.body.message
+    && response.body.message.error_code === 'APP_NOT_INSTALLED') {
+    // If the app is not installed, try to create the check run using GitHub actions token.
+    console.log('Looks like the scanitizer app is not installed on the repo. Falling back to check run creation through GitHub actions app...');
```

Comment:
This is probably not relevant anymore right? Or do we want to still keep this path open?

Reply:
Yeah probably. Let's discuss this once the app thing is done.
Updated #22 to track this.